spec.md

```diff
@@ -841,13 +841,113 @@ For success responses, the following fields are defined:
 | --- | --- | --- |
 | dashboard_url | string | The URL of a web-based management user interface for the Service Instance; we refer to this as a service dashboard. The URL MUST contain enough information for the dashboard to identify the resource being accessed (`9189kdfsk0vfnku` in the example below). Note: a Service Broker that wishes to return `dashboard_url` for a Service Instance MUST return it with the initial response to the provision request, even if the service is provisioned asynchronously. If present, MUST be a non-empty string. |
 | operation | string | For asynchronous responses, Service Brokers MAY return an identifier representing the operation. The value of this field MUST be provided by the Platform with requests to the [Last Operation](#polling-last-operation) endpoint in a percent-encoded query parameter. If present, MUST be a non-empty string. |
+| extension_apis | array-of-objects | For extensions to the Service Broker API, Service Brokers MAY return one or more `extension_api` objects that describe additional API endpoints via an OpenAPI document. See [Extension API Object](#extension-api-object) for more information. |
+
+##### Extension API Object
+
+The `extension_api` object MAY be used to describe any additional endpoint
+to the Open Service Broker API. An example of this could be lifecycle
+management of a Service Instance, (e.g. "Day Two Operations"), like Backup,
+Restore, Stop, Start, Restart and Pause.
```

Comment:
Some of these example operations seem like they would be long-running. Is there any way for a broker to report status of a long running action back to the platform?

Reply:
Related to this, is there a way brokers can provide either success or failure messages for the platform to display? Ideally if the request fails, the UI can give a specific error about what went wrong instead of a generic failed error.

Reply:
Async ops, if enabled, would be defined via the swagger file. The platform could detect and subscribe to ops based on the returned swagger.

Reply:
> Async ops, if enabled, would be defined via the swagger file. The platform could detect and subscribe to ops based on the returned swagger.

I'm not sure how a platform can detect it only from the swagger. A human looking at it might notice an endpoint called `/backup/<id>/status` and guess that it's for checking status of an async operation. But how can the platform know programmatically to do that?

As an end user, I'd be worried if I couldn't check the status of an operation like "Restore" and know that it completed successfully.

I could see it if the platform recognized the `adheres_to` and had special handling for that particular standard.

Reply:
I think the short answer is it's not going to be possible across the board for all APIs.

My thought was that the platform could recognize a callback object in the openapi doc, (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#callbackObject) and other openapi patterns to figure out what it needs, but we won't know until someone tries.

It's the main issue with going the generic route on extensions.